trends_presentation/src/commonMain/kotlin/net/thechance/mena/trends/presentation/screen/user_reel/UserReelViewModel.kt

```diff
@@ -30,14 +31,25 @@ class UserReelViewModel(
     }
 
     override fun onDeleteClick() {
-        // Todo
+        updateState {
+            copy(isConfirmationDialogVisible = true)
+        }
     }
 
     override fun onConfirmDeleteClick() {
-        // Todo
+        viewModelScope.launch {
```

Comment:
You not need for using viewModelScope to provide coroutine scope use tryToExcuteFunction we made to offer this scope .
This code can replace it by : 
 ```suggestion
         tryToExecute(
            block = { reelsRepository.deleteReelById(state.value.id) },
            onSuccess = {
                updateState {
                    copy(isConfirmationDialogVisible = false, isReelDeleted = true)
                }
            },
            onError = { handleError(it) },
        )
```